code-challenges.test.js

```diff
@@ -64,4 +136,40 @@ const accountTransactions2 = [250, -89, 100, -96]
 const accountTransactions3 = []
 // Expected output: []
 
+describe("accumulatedSum", () => {
+  it("returns an array of the accumulating sum", () => {
+    expect(accumulatedSum(accountTransactions1)).toEqual([100, 83, 60, 51])
+    expect(accumulatedSum(accountTransactions2)).toEqual([250, 161, 261, 165])
+    expect(accumulatedSum(accountTransactions3)).toEqual([])
+  })
+})
+
+// ReferenceError: accumulatedSum is not defined
+
 // b) Create the function that makes the test pass.
+
+//Pseudo code:
+  // Create a function named accumulatedSum that takes in an array as an argument with the parameter array
+  // Input: An array of numbers (positive and negative) OR an empty array
+  // Iterate over the array using .reduce() method to accumulate sums of the values in the array starting at the zero index
+  // .reduce() takes in 2 required parameters: accumulator and currentValue and we will also take in the optional index parameter i. The accumlator is the outputted value that we update by adding it with the currentValue
+  // Set initial value to an empty array [], since .reduce() typically just returns one value and we want to output an array of accumulated values
+  // We use spread operator ... to copy the given array so we can use and update the values by adding the currentValue of the iteration to the previous value which we code as 'accumulator[i-1]' 
+  // In order to return an empty array if that's what is given, we have to use the logical OR || to either iterate the summation of the values or just 0 
+  // Output: An array of the accumulating sums or an empty array
+
+const accumulatedSum = (array) => {
+  return array.reduce((accumulator, currentValue, i) => 
+  [...accumulator, currentValue + (accumulator[i-1] || 0)], [])
+}
+
+console.log(accumulatedSum(accountTransactions1)) //  [ 100, 83, 60, 51 ]
+console.log(accumulatedSum(accountTransactions2)) //  [ 250, 161, 261, 165 ]
+console.log(accumulatedSum(accountTransactions3)) //  []
+
+//  accumulatedSum
+// ✓ returns an array of the accumulating sum (7 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       3 passed, 3 total
+// !!!! which is a miracle because this question was sooooo hard :'(
```

Comment:
Amazing job on this one and putting it all together with very detailed pseudocode!!! 👏🏽